crates/nix_health/src/check/direnv.rs

```diff
@@ -93,71 +133,99 @@ fn activation_check(local_flake: &std::path::Path, required: bool) -> Check {
     }
 }
 
-/// Information about a direnv install
-#[derive(Debug, Default, Serialize, Deserialize, Clone)]
-pub struct DirenvInstall {
-    /// Path to the direnv binary
-    pub bin_path: PathBuf,
-
-    /// Contents of `direnvrc`
-    pub direnv_config: Option<String>,
-
-    // bash_path used by direnv
-    pub bash_path: Option<PathBuf>,
-}
-
-impl DirenvInstall {
-    /// Detect user's direnv installation
+/// [Check] if direnv was already allowed in [project_dir]
 
-    pub fn detect() -> anyhow::Result<Self> {
-        let bin_path = which::which("direnv")?;
-        let output = std::process::Command::new(&bin_path)
-            .arg("status")
-            .output()?;
+fn is_direnv_allowed_on(project_dir: &std::path::Path) -> anyhow::Result<bool> {
+    let output = std::process::Command::new("direnv")
+        .args(["status", "--json"])
+        .current_dir(project_dir)
+        .output()?;
+    if output.status.success() {
         let out = String::from_utf8_lossy(&output.stdout);
-        let mut bash_path = None;
-        let mut direnv_config = None;
-        // NOTE: One day we'll switch to using JSON output
-        // https://github.com/direnv/direnv/pull/1142
-        for line in out.lines() {
-            if let Some(path) = line.strip_prefix("bash_path ") {
-                bash_path = Some(PathBuf::from(path));
-            }
-            if let Some(config_dir) = line.strip_prefix("DIRENV_CONFIG ") {
-                let config_file = PathBuf::from(config_dir).join("direnvrc");
-                // Read config_file and assign to direnv_config
-                if config_file.exists() {
-                    let config = std::fs::read_to_string(config_file)?;
-                    direnv_config = Some(config);
-                }
-            }
-        }
-        Ok(Self {
-            bin_path,
-            direnv_config,
-            bash_path,
-        })
+        let status = DirenvStatus::from_json(&out)?;
+        Ok(status.state.is_found_rc_allowed())
+    } else {
+        anyhow::bail!("Unable to run direnv status --json: {:?}", output.stderr)
     }
 }
 
-/// Check if direnv was already activated in [project_dir]
-
-pub fn is_direnv_active_on(project_dir: &std::path::Path) -> anyhow::Result<bool> {
+fn direnv_version() -> anyhow::Result<String> {
```

Comment:
I think this should return `Semver` instead of `String`. 

Function interface should be as smaller as possible.

Reply:
Likewise, this should probably be a method of `DirenvInstall`.